recipes/miniaudio/all/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+cmake_minimum_required(VERSION 3.4)
+project(miniaudio C)
+
+include(GNUInstallDirs)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+if(MSVC OR MSVC90 OR MSVC10)
+    set(MSVC ON)
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
```

Comment:
- Hide all symbols
- at build time: define MA_DLL & MINIAUDIO_IMPLEMENTATION if shared
- at consume time: define MA_DLL

see https://github.com/mackron/miniaudio/blob/d06d4983d3ff512690eb37f5edd25bf96f8f3764/extras/miniaudio_split/miniaudio.h#L149-L177

Reply:
This (as most of the CMakeLists.txt) was taken from the example cited in the docs, bzip2. I just assumed it was good CMake practice and did not think about it tbh.
Is this something that's actually needed (on windows ?) at all ?

Reply:
It's different for bzip2. bzip2 library doesn't handle symbol export at all. miniaudio provides macros to properly export symbols (see link above).

Reply:
`MINIAUDIO_IMPLEMENTATION` is expected to be defined by the consumer to enable the lib implementation (once), when using it in header-only. So I think you're right, cf https://miniaud.io/docs/manual/index.html#Building